### PR TITLE
Redesign and allow people to upload in the multiple locations

### PR DESCRIPTION
**Pull request by NastuzziSamy**
Hey !

I worked on it, for a first time allow people to choose and to send in the same time on multiple locations.

I hope you will enjoy what i tried to do and I am working to manage file transfers via database for a further version

### CODE REVIEW

**Comment by NastuzziSamy** [comment]
Just a little preview ! (sorry, it is in French)

flowupload

**Comment by NastuzziSamy** [comment]
Just a little preview @e-alfred ! (sorry, it is in French)

flowupload

**Comment by e-alfred** [comment]
Thanks for the PR! I will have to review it and merge it as soon as possible.

**Comment by NastuzziSamy** [comment]
Hey @e-alfred,

When do you think you will merge my PR so I can work on the db integration :)

File: ajax/locations.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<?php
+  // Restrict access // ToDo: Enabled for current user?
+  if (!\OCP\User::isLoggedIn() || !\OCP\App::isEnabled('flowupload')) {
```
Comment (e-alfred) [inline ajax/locations.php]: We will have to change this function because both calls are deprecated.

File: templates/main.php [hunk 2]
```diff
@@ -1,58 +1,94 @@
...
+  			<div id="files-setting-showhidden">
+  				<input class="checkbox" id="showhiddenfilesToggle" checked="checked" type="checkbox">
+  				<label for="showhiddenfilesToggle"></label>
+  			</div>
+  			<label for="webdavurl">WebDAV</label>
+  			<input id="webdavurl" readonly="readonly" value="https://cloud.nastuzzi.fr/remote.php/webdav/" type="text">
+  			<em>Utilisez cette adresse pour <a href="https://docs.nextcloud.com/server/12/go.php?to=user-webdav" target="_blank" rel="noreferrer noopener">accéder à vos fichiers par WebDAV</a></em>
+  		</div>
+  	</div>
+  </div>
```
Comment (e-alfred) [inline templates/main.php]: I think this passage is not necessary at all and it contains your personal cloud link too.

**Comment by e-alfred** [comment]
I took a look at your changes and made and found a few things you should change. Additionally maybe a drop down menu would be useful to select an existing folder in the destination folder field.

@Larsene @CHazz @mzary Maybe you can help us with translating the all the additional lines? Any help would be greatly appreciated.

**Comment by e-alfred** [comment]
@NastuzziSamy Any updates on this? I would love to merge but there are a few things that should be resolved. Thanks for your help!

**Comment by NastuzziSamy** [comment]
@e-alfred Everything is fixed like you asked :) 

I can try to free some time so we can add a db (to save locations used by each user + save params like default locations if you are on !)

**Comment by e-alfred** [comment]
I found a few problems regarding support for Nextcloud 14 and upwards, please see the review comments.

**Comment by NastuzziSamy** [comment]
@e-alfred Which comment ? I already fixed them ^^

**Comment by e-alfred** [comment]
@NastuzziSamy The \OC_Response::setStatus calls which were deprecated in Nextcoud 14 and up.

**Comment by e-alfred** [comment]
I would love to merge your PR if you could fix those deprecated calls so it can be used with Nextcloud 14 and 15.

**Comment by NastuzziSamy** [comment]
@e-alfred, sorry for being so late. I commit this week end. Feel free to do whatever you want if I don't do it. Really sorry

**Comment by NastuzziSamy** [comment]
@e-alfred I am confused, did you merged this PR ? ^^

**Comment by e-alfred** [comment]
@NastuzziSamy Any news on your rework? I still like the idea but it still needs some polish. One thing that would be great to have is a folder navigation option when clicking on the "+" button to make it easier for users to select and create folders where files should be uploaded.

**Comment by hezten** [comment]
This is simply taking to long. Is there any way i can move the PR to a fork of the project ? I badly need this feature too :-)

**Comment by e-alfred** [comment]
@hezten If you want to finish this feature it would be awesome! To restart work on this PR you could just fork the flowupload repository of @NastuzziSamy from here:

https://github.com/NastuzziSamy/flowupload

Please make a new PR as soon as you finish your work so it can become part of the official release of Flowupload!

**Comment by Jon089** [comment]
Has this become an official release by any chance?

**Comment by TessyPowder** [comment]
Because so many things changed in master i made another PR with the Changes from NastuzziSamy but in the latest flowupload version.

**Comment by e-alfred** [comment]
@TessyPowder Thanks for your help! Maybe we can close this PR if your new PR is good enough for merging?

**Comment by TessyPowder** [comment]
> @TessyPowder Thanks for your help! Maybe we can close this PR if your new PR is good enough for merging?

Yes. That's probably a good idea

**Comment by NastuzziSamy** [comment]
Hey guys, just saw what you did. 

Thanks for taking your time finishing what I started, too busy this last years :/

Closing this :)

**Comment by NastuzziSamy** [comment]
Actually, is this PR useless or I can rework on it ?

**Comment by NastuzziSamy** [comment]
Do not hesitate to personnaly ping me

**Comment by TessyPowder** [comment]
I am currently working on a implementation of this feature so this PR is probably not necessary anymore.

**Comment by NastuzziSamy** [comment]
Can you then at least co-author me @TessyPowder in your commits ? I saw your PRS, where you are referencing me but not giving me credit 😊

**Comment by TessyPowder** [comment]
> Can you then at least co-author me @TessyPowder in your commits ? I saw your PRS, where you are referencing me but not giving me credit 😊

Ok, how should i do that ?

**Comment by e-alfred** [comment]
PR #73 is fixing this feature finally, thanks for your help still!

**Pull request closed without merging**